UnicornRender/video/include/unicorn/video/CameraFpsController.hpp

```diff
@@ -4,38 +4,50 @@
 * (http://opensource.org/licenses/MIT)
 */
 
-#ifndef UNICORN_VIDEO_CAMERA_FPS_CONTROLLER_HPP
-#define UNICORN_VIDEO_CAMERA_FPS_CONTROLLER_HPP
+#ifndef UNICORN_VIDEO_CAMERAFPSCONTROLLER_HPP
+#define UNICORN_VIDEO_CAMERAFPSCONTROLLER_HPP
 
 #include <unicorn/utility/SharedMacros.hpp>
-#include <unicorn/video/Camera.hpp>
+#include <unicorn/video/Transform.hpp>
 
 namespace unicorn
 {
 namespace video
 {
-/**
- * @brief FPS style camera controller
- */
-class CameraFpsController
+
+/** @brief FPS style camera controller */
+class CameraFpsController : public Transform
 {
 public:
-    UNICORN_EXPORT CameraFpsController(unicorn::video::Camera& camera);
-    UNICORN_EXPORT void MoveUp(float deltaTime) const;
-    UNICORN_EXPORT void MoveDown(float deltaTime) const;
-    UNICORN_EXPORT void MoveLeft(float deltaTime) const;
-    UNICORN_EXPORT void MoveRight(float deltaTime) const;
-    UNICORN_EXPORT void MoveForward(float deltaTime) const;
-    UNICORN_EXPORT void MoveBackward(float deltaTime) const;
-    UNICORN_EXPORT void UpdateView(double posX, double posY);
-    UNICORN_EXPORT void Scroll(float yoffset) const;
-    float sensitivity, speed;
+    /**
+     * @brief Constructs camera fps controller
+     *
+     * @param[out] cameraView reference to camera view
+     */
+    UNICORN_EXPORT CameraFpsController(glm::mat4& cameraView);
+
+    /** @brief Updates view matrix by taking diff between past and new mouse coordinates */
+    UNICORN_EXPORT void UpdateView(float x, float y);
+
+    /** @brief Sets mouse coordinates without updating view matrix */
+    UNICORN_EXPORT void SetViewPositions(float x, float y);
+
+    /** @brief Sets flag to update mouse position without recalculation of orientation
+     * in next UpdateView call */
+    UNICORN_EXPORT void ResetView();
+
+    UNICORN_EXPORT void Update();
 private:
-    double m_lastX, m_lastY, m_yaw, m_pitch;
-    bool m_dirty;
-    unicorn::video::Camera& m_camera;
+    /** @brief Calculates orientation in space */
+    virtual void UpdateOrientation() override;
+
+    glm::vec2 m_mousePosition;
+    bool m_dirtyViewPosition;
+
+    glm::mat4& m_cameraView;
```

Comment:
What's the logic behind this class not being copy constructible? It does not look like it manages any resources.